include/zephyr/sys/util.h

```diff
@@ -538,7 +538,10 @@ size_t hex2bin(const char *hex, size_t hexlen, uint8_t *buf, size_t buflen);
  */
 static inline uint8_t bcd2bin(uint8_t bcd)
 {
-	return ((10 * (bcd >> 4)) + (bcd & 0x0F));
```

Comment:
What was the compiler warning here?  There's no truncation and as I read it all the math here works fine without overflow in either int or unsigned char precision.  If it was just whining that you're returning a (promoted) int type as uint8_t, put a cast on the result.